app/dao/fact_notification_status_dao.py

```diff
@@ -239,7 +239,7 @@ def fetch_notification_status_for_service_for_day(bst_day, service_id):
 
 
 def fetch_notification_status_for_service_for_today_and_7_previous_days(service_id, by_template=False, limit_days=7):
-    start_date = utc_midnight_n_days_ago(limit_days - 1)
+    start_date = utc_midnight_n_days_ago(limit_days)
 
     stats_for_7_days = db.session.query(
```

Comment:
An idea that crossed my mind to make this code a bit more testable is pulling the `stats_for_7_days` and `stats_for_today` out into their own methods. We could then reuse them in other contexts in the future and get a bit more granular and focused with our testing considering we have had repeated issues in this area of the code.

Reply:
I like that idea, if you want to do it in this PR feel free